packages/pie-toolbox/src/code/editable-html/editor.jsx

```diff
@@ -420,18 +421,74 @@ export class Editor extends React.Component {
   };
 
   onEditingDone = () => {
-    if (this.state.isHtmlMode) {
+    const { isHtmlMode, dialog, value, pendingImages } = this.state;
+
+    // Handling HTML mode and dialog state
+    if (isHtmlMode && !dialog?.open) {
+      const currentValue = htmlToValue(value.document.text);
+      const previewText = this.renderPreviewText(currentValue.document.text);
+
+      this.openDialogWithPreview(currentValue, previewText);
       return;
     }
 
-    const { pendingImages } = this.state;
+    // Early return if HTML mode is enabled
+    if (isHtmlMode) return;
 
+    // Handling pending images
     if (pendingImages.length) {
-      this.setState({ scheduled: true });
+      this.scheduleImageProcessing();
       return;
     }
 
-    log('[onEditingDone]');
+    // Finalizing editing
+    this.finalizeEditing();
+  };
+
+  renderPreviewText = (text) => {
+    const { classes } = this.props;
+    return (
+      <div>
+        <div className={classes.previewSeparator}></div>
+        <div className={classes.previewText}>{text}</div>
+        <div className={classes.previewSeparator} style={{ marginTop: '10px' }}></div>
+        <div>Would you like to save these changes ?</div>
+      </div>
+    );
+  };
+
+  openDialogWithPreview = (currentValue, previewText) => {
+    this.setState({
+      dialog: {
+        open: true,
+        title: 'Preview of edited HTML',
+        text: previewText,
+        onConfirmText: 'Save changes',
+        onCloseText: 'Continue editing',
+        onConfirm: () => {
+          this.handleConfirm(currentValue);
+        },
+        onClose: this.handleCloseDialog,
+      },
+    });
+  };
+
+  handleConfirm = (currentValue) => {
+    this.setState({ value: currentValue });
```

Comment:
I would rename this to be html mode specific and add a comment about what its behaviour.